source/extensions/matching/input_matchers/cel_matcher/matcher.cc

```diff
@@ -0,0 +1,50 @@
+#include "source/extensions/matching/input_matchers/cel_matcher/matcher.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Matching {
+namespace InputMatchers {
+namespace CelMatcher {
+
+CelInputMatcher::CelInputMatcher(const CelExpression& input_expr) {
+  expr_builder_ = Extensions::Filters::Common::Expr::createBuilder(nullptr);
```

Comment:
Builders are expensive - they are essentially "compiler instances". They need to be defined and reused in the factories - see other examples how it's done.

Reply:
yea, I created it in factory in previous revision but changed it to sit along with expression in most recent revision to make them in one place.

I agreed it better to set in factory. Will update.